provider/resource_keycloak_group_roles.go

```diff
@@ -135,12 +152,16 @@ func resourceKeycloakGroupRolesRead(data *schema.ResourceData, meta interface{})
 	var roleIds []string
 
 	for _, realmRole := range roles.RealmMappings {
-		roleIds = append(roleIds, realmRole.Id)
+		if data.Get("exhaustive").(bool) || containsAString(sortedRoleIds, realmRole.Id) {
+			roleIds = append(roleIds, realmRole.Id)
+		}
 	}
 
 	for _, clientRoleMapping := range roles.ClientMappings {
 		for _, clientRole := range clientRoleMapping.Mappings {
-			roleIds = append(roleIds, clientRole.Id)
+			if data.Get("exhaustive").(bool) || containsAString(sortedRoleIds, clientRole.Id) {
```

Comment:
I'd pull this `data.Get` invocation and type assertion out to its own variable, so it can be used here as well as on line 155.  just saves us some extra CPU cycles.

Reply:
You are right. I do it

Reply:
@mrparkers : done